src/Core/Form/IdentifiableObject/DataProvider/CategoryFormDataProvider.php

```diff
@@ -82,6 +100,47 @@ public function getData($categoryId)
         /** @var EditableCategory $editableCategory */
         $editableCategory = $this->queryBus->handle(new GetCategoryForEditing($categoryId));
 
+        $coverImages = $thumbnailImages = null;
+        $categoryId = (int) $categoryId;
+        $categoryUrl = $this->categoryProvider->getUrl($categoryId, '{friendly-url}');
+        $coverImage = $editableCategory->getCoverImage();
+        if ($coverImage) {
+            $coverImages[] = [
```

Comment:
Are you sure you can consider `$coverImages` as an array and push a value in it? Because its initial value is `null`

Also I'm not sure you need an array of coverImages here since there can be only one, so maybe this could be enough:
```suggestion
            $coverImage = [
```
It's simply a flatten array for one image

Reply:
Ah unless you need to always provide a list of images because of the form type used for these three elements? Then I understand but then maybe `$coverImages` should be initialized with empty array instead of null?

Reply:
Yeah exactly, I need to work with menu thumbnails(which can be multiple), so cover and thumbnail are arrays if they consist only of single element.

Reply:
No no I get it better now, it's good this ay!